### DISCUSSION TITLE
defmain is not working

### DISCUSSION THREAD

**Discussion by dineshbhardwaj** [hylang/hy#2489] [Q&A] [answered]
Hi , 
In older version my code like 
(defmain [] (print "hi")) was working. now it does not work. it says "NameError: name 'defmain' is not defined". 
what is the new equalent for main function in hylang?

**Comment by Kodiologist** [comment]
Hy releases are in general not compatible with each other. See NEWS for each breaking change and how to address it, or downgrade to your preferred version of Hy.

**Reply by dineshbhardwaj** [reply, marked as answer]
Thanks for quick reply. so, I need to add just this line: 
(require hyrule [defmain])

**Reply by Kodiologist** [reply]
Yes, that sounds correct.
